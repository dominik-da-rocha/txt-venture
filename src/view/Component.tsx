import React from "react";

export interface ComponentProps {
  id: string;
  children?: React.ReactNode;
  key?: string;
  className?: string;
}

export interface ComponentState {}

export class Component<
  P extends ComponentProps,
  S extends ComponentState
> extends React.Component<P, S> {
  constructor(props: P, state: S) {
    super(props);
    this.state = state;
  }

  render() {
    return (
      <div id={this.props.id} className={this.className()}>
        {this.props.children}
      </div>
    );
  }

  makeId(id: string): string {
    return this.props.id + "-" + id;
  }

  className(name?: string) {
    let names: string[] = ["Component"];
    if (name) {
      names.push(name);
    }
    if (this.props.className) {
      names.push(this.props.className as string);
    }
    return names.join(" ");
  }
}

export default Component;
