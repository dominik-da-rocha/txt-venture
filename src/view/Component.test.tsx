import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Renderer from "react-test-renderer";
import React from "react";
import Component, { ComponentProps, ComponentState } from "./Component";

Enzyme.configure({ adapter: new Adapter() });

class TestComponent extends Component<ComponentProps, ComponentState> {
  render() {
    return (
      <div id={this.props.id}>
        <div id={this.makeId("child1")}></div>
        <div id={this.makeId("child2")}></div>
      </div>
    );
  }
}

describe("Component", () => {
  it("renders without crashing", () => {
    expect(() => {
      shallow(<Component id="component" />);
    }).not.toThrow();
  });

  it("renders snapshot", () => {
    const tree = Renderer.create(
      <Component id="test" className="Test" key="test" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders className Component", () => {
    let wrapper = shallow(<Component id="component" />);
    expect(wrapper.hasClass("Component")).toBeTruthy();
  });

  it("renders find by id", () => {
    let wrapper = shallow(<Component id="myId" />);
    expect(wrapper.find("#myId")).toHaveLength(1);
  });

  it("renders className defined by user", () => {
    let wrapper = shallow(<Component id="component" className="Fancy" />);
    expect(wrapper.hasClass("Fancy")).toBeTruthy();
  });

  it("renders id", () => {
    let wrapper = shallow(<TestComponent id="parent"></TestComponent>);
    expect(wrapper.find("#parent-child1")).toHaveLength(1);
    expect(wrapper.find("#parent-child2")).toHaveLength(1);
  });
});
