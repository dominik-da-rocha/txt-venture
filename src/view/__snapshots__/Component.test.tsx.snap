// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders snapshot 1`] = `
<div
  className="Component Test"
  id="test"
/>
`;
